OpenRA.Mods.Common/Traits/Render/RenderDetectionCircle.cs

```diff
@@ -76,8 +83,20 @@ IEnumerable<IRenderable> IRenderAnnotationsWhenSelected.RenderAnnotations(Actor
 				info.BorderWidth);
 		}
 
+		IEnumerable<IRenderable> IRenderAnnotationsWhenSelected.RenderAnnotations(Actor self, WorldRenderer wr)
+		{
+			return RenderCircle(self, wr, DetectionCircleVisibility.WhenSelected);
+		}
+
 		bool IRenderAnnotationsWhenSelected.SpatiallyPartitionable => false;
 
+		IEnumerable<IRenderable> IRenderAnnotations.RenderAnnotations(Actor self, WorldRenderer wr)
+		{
+			return RenderCircle(self, wr, DetectionCircleVisibility.Always);
+		}
+
+		bool IRenderAnnotations.SpatiallyPartitionable => false;
```

Comment:
Just throwing it in here: We should get around to enabling spatial partitioning on any kind of circles at some point, right now we're rendering any visible circles all the time, regardless of how far they are from the viewport.